login/logout.go

```diff
@@ -0,0 +1,79 @@
+package login
+
+import (
+	"net/url"
+	"regexp"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/jsonapi"
+	"github.com/almighty/almighty-core/log"
+	"github.com/goadesign/goa"
+)
+
+// KeycloakLogoutService represents a keyclaok logout service
+type KeycloakLogoutService struct {
+}
+
+// LogoutService represents logout service interface
+type LogoutService interface {
+	Logout(ctx *app.LogoutLogoutContext, logoutEndpoint string, validRedirectURL string) error
+}
+
+// Logout logs out user
+func (s *KeycloakLogoutService) Logout(ctx *app.LogoutLogoutContext, logoutEndpoint string, validRedirectURL string) error {
+	redirect := ctx.Redirect
+	referrer := ctx.RequestData.Header.Get("Referer")
+	if redirect == nil {
+		if referrer == "" {
+			log.Error(ctx, map[string]interface{}{}, "Failed to logout. Referer Header and redirect param are both empty.")
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrBadRequest("Referer Header and redirect param are both empty. At least one should be specified."))
```

Comment:
Not sure tbh how this message should look? There are two sentences in one line:
"Referer Header and redirect param are both empty. At least one should be specified."